docs/docs/module_guides/observability/instrumentation.md

```diff
@@ -185,20 +186,36 @@ class MyCustomSpanHandler(BaseSpanHandler[MyCustomSpan]):
         return "MyCustomSpanHandler"
 
     def new_span(
-        self, id: str, parent_span_id: Optional[str], **kwargs
+        self,
+        id_: str,
+        bound_args: inspect.BoundArguments,
+        instance: Optional[Any] = None,
+        parent_span_id: Optional[str] = None,
+        tags: Dict[str, Any] = {},
```

Comment:
ah, weird things can happen if you set a kwarg to default to a dict

Instead we should have `Optional[Dict[str, Any]] = None` and then convert to an empty dict in the function when needed

Reply:
Agreed, updated!